src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SvCigarUtils.java

```diff
@@ -46,9 +48,11 @@ public static int getNumClippedBases(final boolean fromStart, final Cigar cigar)
      * from the start or from the end
      * @param fromStart     from the start of the template or not
      * @param cigarElements the ordered {@link CigarElement}'s of a cigar
+     * @throws              IllegalArgumentException if the input cigar is null or empty
      */
     @VisibleForTesting
     public static int getNumClippedBases(final boolean fromStart, final List<CigarElement> cigarElements) {
```

Comment:
Would it make sense to simply return 0 for an empty Cigar element list? Actually are empty Cigar legal, if so 0 could be acceptable perhaps even desirable.

Reply:
My understanding is that most likely this is an error on the user side hence I'd want the user know.
Because, what does empty cigar list mean?
It could be legal to be empty, but I've yet to see it, and again like I said, most likely an error.

I'm definitely open to change the behavior if there's a valid and concrete example showing empty cigar list.

Reply:
Based on the `Cigar` class in htsjdk, empty CIGARs are possible.

On the other hand, I would ask the question why this method be invoked on such CIGARs?

Unmapped records have empty CIGARs for sure, but this gets to the design choice: whether it is the caller's responsibility to check if the record is mapped before querying such properties of its CIGAR (hence this method makes the mapped & aligned assumption&mdash;and barks if false), or should this method silently return a 0.